packages/vaex-core/setup.py

```diff
@@ -121,10 +121,13 @@ def __str__(self):
     ]
     extra_compile_args.append("-g")
     extra_compile_args += extra_dev_options
+    # this makes sure we catch error in using pybind11, see https://github.com/vaexio/vaex/issues/2439
+    extra_compile_args += ["-UNDEBUG"]
```

Comment:
I wonder if #2441 will override this one, or whether this one is somehow passed alongside the CXXFLAGS (and which behaviour we end up with when both are merged)

Reply:
because this PR causes cibuildwheel to run with `-UNDEBUG`:
```
creating build/temp.linux-x86_64-cpython-38/src
    g++ -pthread -Wl,-strip-all -fPIC -I/tmp/pip-build-env-p52uv4h5/overlay/lib/python3.8/site-packages/numpy/core/include -I/opt/_internal/cpython-3.8.18/include/python3.8 -c src/vaexfast.cpp -o build/temp.linux-x86_64-cpython-38/src/vaexfast.o -std=c++11 -O3 -funroll-loops -Werror=return-type -Wno-unused-parameter -g -UNDEBUG
```
https://github.com/vaexio/vaex/actions/runs/11233046337/job/31225954823?pr=2440#step:7:1403

Reply:
Yes, this is perfect!